sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -260,6 +260,20 @@ case class Join(
     condition: Option[Expression])
   extends BinaryNode with PredicateHelper {
 
+  private def isBrocastHint(plan: LogicalPlan): Boolean = {
```

Comment:
E.g. each `case Join(left, right, joinType, condition) =>` would need to change to `case Join(left, right, joinType, condition, leftBroadcastHint, rightBroadcastHint)`.